hail/src/main/scala/is/hail/expr/ir/streams/EmitStream.scala

```diff
@@ -257,6 +261,147 @@ object EmitStream {
             })
 
         }
+        
+      case x@StreamBufferedAggregate(streamChild, initAggs, newKey, seqOps, name,
+        aggSignatures: IndexedSeq[PhysicalAggSig]) =>
+        val region = mb.genFieldThisRef[Region]("stream_buff_agg_region")
+        produce(streamChild, cb)
+          .map(cb) { case childStream: SStreamValue =>
+            val childProducer = childStream.producer
+            val eltField = mb.newEmitField("stream_buff_agg_elt", childProducer.element.emitType)
+            val newKeyVType = typeWithReqx(newKey)
+            val kb = mb.ecb
+            val nestedStates = aggSignatures.toArray.map(sig => AggStateSig.getState(sig.state, kb))
+            val nested = StateTuple(nestedStates)
+            val dictState = new DictState(kb, newKeyVType, nested)
+            val maxSize = mb.genFieldThisRef[Int]("stream_buff_agg_max_size")
+            val nodeArray = mb.genFieldThisRef[Array[Long]]("stream_buff_agg_element_array")
+            val idx = mb.genFieldThisRef[Int]("stream_buff_agg_idx")
+            val returnStreamType= x.typ.asInstanceOf[TStream]
+            val returnElemType = returnStreamType.elementType
+            val tupleFieldTypes = aggSignatures.map(_ => TBinary)
+            val tupleFields = (0 to tupleFieldTypes.length).zip(tupleFieldTypes).map { case (fieldIdx, fieldType) => TupleField(fieldIdx, fieldType) }
+            val serializedAggSType = SStackStruct(TTuple(tupleFields), tupleFieldTypes.map(_ => EmitType(SBinaryPointer(PCanonicalBinary()), true)).toIndexedSeq)
+            val keyAndAggFields = newKeyVType.canonicalPType.asInstanceOf[PCanonicalStruct].sType.fieldEmitTypes :+ EmitType(serializedAggSType, true) // TODO is this required?
```

Comment:
yes, required is right. We never generate a record with a key and missing agg tuple. Can remove the todo.